azurelinuxagent/ga/exthandlers.py

```diff
@@ -315,7 +316,7 @@ def handle_ext_handlers(self, etag=None):
             logger.verbose("No extension handler config found")
             return
 
-        if conf.get_enable_overprovisioning():
+        if conf.get_enable_overprovisioning() and not isinstance(self.protocol, MetadataProtocol):
```

Comment:
maybe instead add a supports_overprovisioning() method to the protocol class? and a warning if the setting is true but the protocol doesn't support it? thanks

Reply:
@narrieta Just updated with two commits to cover this.  Let me know what you think.
I debated having the base class raise NotImplemented, but didn't want to have to go to every subclass and override it.  Instead opted for the "everyone supports unless explicitly setting not".  Happy to change it if you prefer

Reply:
thanks @adepue, I think your approach is fine. can add a warning if the setting is true, but the protocol doesn't support it?

After that i can run our automation on your PR. On your side, there is a makepkg.py script on the root of the repo that will create a package for the agent. You can test that package to see if there are any other issues (you need to sudo to execute the script).

Reply:
@narrieta Updated to log a warning.  Will test the package on this side as well.